File: YDAppleKit/YDObjcKit/YDKit/YDSingleton/YDSingletonPattern.m
//
//  YDSingletonPattern.m
//  YDKit
//
//  Created by rowena on 2018/7/27.
//  Copyright © 2018年 王允顶. All rights reserved.
//

#import "YDSingletonPattern.h"

@implementation YDSingletonPattern

static YDSingletonPattern *instance = nil;

/**
 * 创建单例
 */
+ (instancetype)sharedInstance {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        instance = [[YDSingletonPattern alloc] init];
    });
    return instance;
}

+ (instancetype)allocWithZone:(struct _NSZone *)zone {
    if (instance == nil) {
        static dispatch_once_t onceToken;
        dispatch_once(&onceToken, ^{
            instance = [super allocWithZone:zone];
        });
    }
    return instance;
}

@end

File: YDAppleKit/YDSwiftKit/YDKit/YDSingletonPattern/YDSingletonPattern.swift
//
//  YDSingletonPattern.swift
//  YDKit
//
//  Created by 王允顶 on 2018/7/31.
//  Copyright © 2018年 王允顶. All rights reserved.
//

import UIKit

final class YDSingletonPattern: NSObject {

    private static let instance: YDSingletonPattern = YDSingletonPattern()
    class func sharedInstance() -> YDSingletonPattern {
        return instance
    }
    
    //将构成方法私有化
    private override init() {
        
    }
    
    
    
}
